dig.go

```diff
@@ -220,7 +231,18 @@ func (c *Container) remove(nodes []node) {
 func (c *Container) constructorArgs(ctype reflect.Type) ([]reflect.Value, error) {
 	args := make([]reflect.Value, 0, ctype.NumIn())
 	for i := 0; i < ctype.NumIn(); i++ {
-		arg, err := c.get(ctype.In(i))
+		var (
+			arg reflect.Value
+			err error
+		)
+
+		t := ctype.In(i)
+		if t.Implements(_parameterObjectType) {
+			arg, err = c.getParameterObject(t)
```

Comment:
nit: getting parameterObject can be part of `c.get(t)`. Then you don't need to declare variables and assign. implementation wise we just want an object back from container.